File: Python/sentimental-credit/credit.py
# TODO
from cs50 import get_string
from cs50 import get_int

while True:
    try:
        card = get_int("the card number is ")
    except ValueError:
        print("it must be all numbers")
#    if len(card) > 16 or len(card) < 13:
#        print("it must contain between 13 and 16 digits ")
    else:
        break
tipe = 0  # to see which kind of card it could be
x = 0  # to see how many numbers are in the card and which numbers are even or odd
mult = 0  # initialising to later sum the multiplications
mult = int(mult)  # setting to int to truncate the values
sum = 0  # initialising to later sum the sum
sum = int(sum)  # setting to int to truncate the values
while card >= 1:  # till we get to the last digit
    x += 1
    if x % 2 == 0:  # if even, mult by two than sum to the mult
        mult = mult + int(((2 * (card % 10)) / 10) % 10) + int(2 * (card % 10) % 10)
    else:  # if odd, just sum to the sum
        sum = sum + int(card % 10)
    card = card // 10  # removes the last digit of the car
# if the first two digits are 34 or 37, it's a AMEX card
    if card == 34 or card == 37:
        tipe = 1
# if the first two digitas are between 51 and 55, it's a MASTERCARD
    if card >= 51 and card <= 55:
        tipe = 2
# if the first digit is 4, it's a Visa card
    if card == 4:
        tipe = 3
total = sum + mult
lastdig = total % 10
# seeing if the card is valid and AMEX
if lastdig == 0 and x == 15 and tipe == 1:
    print("AMEX")
# seeing if the card is valid and MASTERCARD
elif lastdig == 0 and x == 16 and tipe == 2:
    print("MASTERCARD")
# seeing if the card is valid and VISA
elif lastdig == 0 and x == 13 and tipe == 3:
    print("VISA")
# seeing if the card is valid and VISA
elif lastdig == 0 and x == 16 and tipe == 3:
    print("VISA")
# if it's not one of the above, it's INVALID
else:
    print("INVALID")

File: C/credit/credit.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    long tipe = 0; //To show which kind of card this is, if valid.
    long sum = 0; // to make the sum of the non multiplied numbers
    long mult = 0; // to make the sum of the multipled numbers
    long card = get_long("Card numer ? "); // asking the card's number
    int x = 0; // to count how many numbers are in the card
    while (card >= 1) // while the total card number, being divided by 10, isn't lower than 1
    {
        x++; // counting numbers of digits
        if (x % 2 == 0) // if the number is odd, thus is a other number starting from second last
            {
            mult = mult + (((2 * ( card % 10)) / 10) % 10) + ( 2 * (card % 10) % 10) ;
             // multply it by two than some the first and last digit of it to the mult sum
            }
        else // if it's not a number to be multiplied
            {
            sum = sum + card % 10; // sums it to the sum of numbers that aren't multiplied

            }
        card = card / 10; //removes the used number and continues

        if (card == 34 || card == 37) {tipe = 1;} // if the first two digitas are 34 or 37, it's a AMEX card
        if (card >= 51 && card <= 55) {tipe = 2;}// if the first two digitas are between 51 and 55, it's a MASTERCARD
        if (card == 4 ) { tipe = 3;}// if the first digit is 4, it's a Visa card
    }
    long total = sum + mult; //summing the sum of the multiplied and non multipled
    int lastdig = total % 10; // seeing if the last digit of the sum is 0
    if(lastdig == 0 && x == 15 && tipe == 1)
        {printf("AMEX\n");} // seeing if the card is valid and AMEX
    else if(lastdig == 0 && x == 16 && tipe == 2)
        {printf("MASTERCARD\n");}// seeing if the card is valid and MASTERCARD
    else if(lastdig == 0 && x == 13 && tipe == 3)
        {printf("VISA\n");}// seeing if the card is valid and VISA
    else if(lastdig == 0 && x== 16 && tipe == 3)
        {printf("VISA\n");}// seeing if the card is valid and VISA
    else {printf("INVALID\n");}// if it's not one of the above, it's INVALID
}